apps/meteor/app/file-upload/server/methods/sendFileMessage.ts

```diff
@@ -22,13 +24,48 @@ import { FileUpload } from '../lib/FileUpload';
 
 function validateFileRequiredFields(file: Partial<IUpload>): asserts file is AtLeast<IUpload, '_id' | 'name' | 'type' | 'size'> {
 	const requiredFields = ['_id', 'name', 'type', 'size'];
-	requiredFields.forEach((field) => {
+	for (const field of requiredFields) {
 		if (!Object.keys(file).includes(field)) {
 			throw new Meteor.Error('error-invalid-file', 'Invalid file');
 		}
-	});
+	}
 }
 
+const logger = new Logger('sendFileMessage');
+
+type parseMultipleFilesIntoMessageAttachmentsResult = { files: FileProp[]; attachments: MessageAttachment[] };
+export const parseMultipleFilesIntoMessageAttachments = async (
+	filesToConfirm: Partial<IUpload>[],
+	roomId: string,
+	user: IUser,
+): Promise<parseMultipleFilesIntoMessageAttachmentsResult> => {
+	const result: parseMultipleFilesIntoMessageAttachmentsResult = { files: [], attachments: [] };
+
+	await Promise.all(
+		filesToConfirm.reduce<Array<Promise<void>>>((acc, file) => {
```

Comment:
i think this is a good case for `array.map` instead  `reduce`

Reply:
It was a map, then it needs another loop to filter out possible empty data, so I changed it to a single reduce

Reply:
The reduce was only being used as a loop (it was pushing promises into an array individually); Even if the performance change here was relevant, a reduce with several `array.push` would hardly be better than a map+filter.

The filter turned out to be unnecessary anyway, so I changed this into a regular map.